lib/ldclient-rb/ldclient.rb

```diff
@@ -56,19 +56,23 @@ def flush
       end
 
       if !events.empty?
-        res = log_timings("Flush events") {
-          next @client.post (@config.base_uri + "/api/events/bulk") do |req|
-            req.headers['Authorization'] = 'api_key ' + @api_key
-            req.headers['User-Agent'] = 'RubyClient/' + LaunchDarkly::VERSION
-            req.headers['Content-Type'] = 'application/json'
-            req.body = events.to_json
-            req.options.timeout = @config.read_timeout
-            req.options.open_timeout = @config.connect_timeout
-          end
-        }
-        if res.status != 200
-          @config.logger.error("[LDClient] Unexpected status code while processing events: #{res.status}")
+        post_flushed_events(events)
+      end
+    end
+
+    def post_flushed_events(events)
+      res = log_timings("Flush events") {
+        next @client.post (@config.base_uri + "/api/events/bulk") do |req|
+          req.headers['Authorization'] = 'api_key ' + @api_key
+          req.headers['User-Agent'] = 'RubyClient/' + LaunchDarkly::VERSION
+          req.headers['Content-Type'] = 'application/json'
+          req.body = events.to_json
+          req.options.timeout = @config.read_timeout
+          req.options.open_timeout = @config.connect_timeout
         end
+      }
+      if res.status != 200
```

Comment:
I changed this to `res.status/100 != 2` since I assume `100 Continue` responses are also not accepted.  I also fixed 2 other places where it was checking for `200` specifically.